client/state/sharing/publicize/publicize-actions/reducer.js

```diff
@@ -53,6 +57,11 @@ export const scheduled = createReducer( {}, {
 		),
 	[ PUBLICIZE_SHARE_ACTION_EDIT_SUCCESS ]:
 		( state, { siteId, postId, item } ) => updateDataForPost( item, state, siteId, postId, item.ID ),
+	[ PUBLICIZE_SHARE_ACTION_SCHEDULE_SUCCESS ]:
+		( state, { siteId, postId, items } ) => {
+			items.forEach( item => state = updateDataForPost( item, state, siteId, postId, item.ID ) );
```

Comment:
Usually yes, but I don't think that is the case. Of course, we can spend time in refactoring `updateDataForPost` so that is 10% more understandable for a person familiar with functional approach, but I do not believe that is a good time investment and am also pretty convinced that over-use of functional approach leads to a code that has much steeper barrier to entry for people not familiar with it.

That being said, I agree with you that we should simplify whenever possible, but I cannot imagine a way of doing so here where end result is better then this and is not invasive into changing every selector in this file because of changing `updateDataForPost` signature.
 
Since, like many selector trees, this one is deeply nested,  `updateDataForPost` updates deeply nested property in the redux tree, under site / post subtrees. on top of that, since actions are indexed by IDs it allows for changing the one with ID that matches.

Since we get arrays not indexed the same way from API, any use of array functions (such as map) is not helpful since we need to reindex IDs first. Doing several passes , reindexing idrs and then using array functions defeats the reason to use them since that does not improve readability in my opinion.